easypost/util.py

```diff
@@ -0,0 +1,36 @@
+from typing import List
+
+from easypost.easypost_object import EasyPostObject
+from easypost.error import Error
+
+
+class Util:
+    @staticmethod
+    def _get_lowest_object_rate(
+        easypost_object: EasyPostObject,
+        carriers: List[str] = None,
+        services: List[str] = None,
+        rates_key: str = "rates",
+    ):
+        """Gets the lowest rate of an EasyPost object such as a Shipment, Order, or Pickup."""
+        carriers = carriers or []
+        services = services or []
+        lowest_rate = None
+
+        carriers = [carrier.lower() for carrier in carriers]
+        services = [service.lower() for service in services]
+
+        for rate in easypost_object[rates_key]:
+            if len(carriers) > 0 and rate.carrier.lower() not in carriers:
```

Comment:
nit: could be more Pythonic and just say `if carriers`, since an empty list == False in Python. Same with `services` below.

Reply:
True. This was just copied over from the original logic to a different file. Will simplify it.